spec/lib/miq_automation_engine/miq_ae_method_dispatch_spec.rb

```diff
@@ -0,0 +1,105 @@
+    @pidfile = File.join(Dir.mktmpdir, "rip_van_winkle.pid")
+    clear_domain
+  end
+
+  def clear_domain
+    MiqAeDomain.find_by_name(@domain).try(:destroy)
+  end
+
+  def rip_van_winkle_script
+    <<-'RUBY'
+      pidfile    = $evm.inputs['pidfile']
+      File.open(pidfile, 'w') { |file| file.write(Process.pid.to_s) }
+      10.times do
+          STDERR.puts "Hello from stderr channel"
+          STDOUT.puts "Hello from stdout channel"
+      end
+      sleep(600)
+    RUBY
+  end
+
+  def std_script
+    <<-'RUBY'
+      pidfile    = $evm.inputs['pidfile']
+      File.open(pidfile, 'w') { |file| file.write(Process.pid.to_s) }
+      $evm.root['method_pid'] = Process.pid
+      10.times do
+          STDERR.puts "Hello from stderr channel"
+          STDOUT.puts "Hello from stdout channel"
+      end
+      exit(MIQ_OK)
+    RUBY
+  end
+
+  def setup_model(method_script)
+    dom = FactoryGirl.create(:miq_ae_domain, :enabled => true, :name => @domain)
+    ns  = FactoryGirl.create(:miq_ae_namespace, :parent_id => dom.id, :name => @namespace)
+    @ns_fqname = ns.fqname
+    create_method_class(:namespace => @ns_fqname, :name => @method_class,
+                        :method_script => method_script)
+    create_root_class(:namespace => @ns_fqname, :name => @root_class)
+  end
+
+  def create_method_class(attrs = {})
+    params = {'pidfile' => {'aetype'        => 'attribute',
+                            'datatype'      => 'string',
+                            'default_value' => @pidfile}
+             }
+    method_script = attrs.delete(:method_script)
+    ae_fields = {'execute' => {:aetype => 'method', :datatype => 'string'}}
+    ae_instances = {@method_instance => {'execute' => {:value => @method_name}}}
+    ae_methods = {@method_name => {:scope => 'instance', :location => 'inline',
+                                   :data => method_script,
+                                   :language => 'ruby', 'params' => params}}
+
+    FactoryGirl.create(:miq_ae_class, :with_instances_and_methods,
+                       attrs.merge('ae_fields'    => ae_fields,
+                                   'ae_instances' => ae_instances,
+                                   'ae_methods'   => ae_methods))
+  end
+
+  def create_root_class(attrs = {})
+    ae_fields = {'rel1' => {:aetype => 'relationship', :datatype => 'string'}}
+    fqname = "/#{@domain}/#{@namespace}/#{@method_class}/#{@method_instance}"
+    ae_instances = {@root_instance => {'rel1' => {:value => fqname}}}
+    FactoryGirl.create(:miq_ae_class, :with_instances_and_methods,
+                       attrs.merge('ae_fields'    => ae_fields,
+                                   'ae_methods'   => {},
+                                   'ae_instances' => ae_instances))
+  end
+
+  it "long running method" do
+    File.delete(@pidfile) if File.exist?(@pidfile)
+    setup_model(rip_van_winkle_script)
+    send_ae_request_via_queue(@automate_args, 2)
```

Comment:
@jrafanie @Fryguy 
This is setting the queue timeout to 2 seconds, it expects that automate request to end in 2 seconds. Since the method is in sleep, it will get terminated by our ensure block and the spec checks to see if the process has been terminated and we haven't left orphan process around.